contracts/hyphen/LiquidityPool.sol

```diff
@@ -480,7 +474,7 @@ contract LiquidityPool is
         );
     }
 
-    function getRewardAmount(uint256 amount, address tokenAddress) public view returns (uint256 rewardAmount) {
+    function getRewardAmount(uint256 amount, address tokenAddress) public view override returns (uint256 rewardAmount) {
         uint256 currentLiquidity = getCurrentLiquidity(tokenAddress);
```

Comment:
Need to check the getCurrentLiquidity method in case it returns a negative value... the deposit transaction would fail. 
If liquidity in pool is very low, and gas fee accumulated + incentive pool is more than available liquidity, getCurrentLiquidity fails in this case.
In this case getCurrentLiquidity should return 0 instead of failing the transaction